### DISCUSSION TITLE
Is there a way to exclude multiple apps using LIKE operator?

### DISCUSSION THREAD

**Discussion by ztrhgf** [Romanitho/Winget-AutoUpdate#537] [Q&A]
Is there a way to exclude multiple apps using LIKE operator? 

I need to exclude a lot of "developer" apps, like  different version of .Net, Python, VS, .... and its impossible to gather all those IDs and maintain them.

So something like LIKE or CONTAINS operator for matching IDs would be nice.

For example:
Microsoft.DotNet.SDK.*
Microsoft.DotNet.Runtime.*
Python.Python.*

**Comment by KnifMelti** [comment]
No, there isn't as of today.

**Comment by ztrhgf** [comment]
It would be very helpful for developer apps like python, visual studio etc
which exists in a lot of separate version I. E. It is quite hard to exclude
them right now.

Not mentioning that such list has to be updated regularly as new versions
are released.

Dne čt 25. 1. 2024 18:41 uživatel Göran Axel Johannesson <
***@***.***> napsal:

**Reply by KnifMelti** [reply]
@Romanitho (the Maintainer of this Repo) has been active just once since 2023-12-13...
...not sure where this is going in the future (hope he and his family are well!)

**Reply by Romanitho** [reply]
Hi :)
I think this is something that can more or less easy to implement. Time is the thing 😅

**Comment by ztrhgf** [comment]
I can create pull request for this if that's fine?

Dne po 19. 2. 2024 14:13 uživatel Romain ***@***.***> napsal:

**Reply by Romanitho** [reply]
Sure. You'll be credited ;)

**Comment by ztrhgf** [comment]
Just asking because there is my another push request waiting 🙂
https://github.com/Romanitho/Winget-AutoUpdate/pull/541

Anyway, I will try to code this one ASAP.

Dne út 20. 2. 2024 19:22 uživatel Romain ***@***.***> napsal:

**Reply by Romanitho** [reply]
The #541 still need to be discussed 😅 as it is specific and complxe case (that might soon be managed by winget if I don't say wrong.)

**Comment by ztrhgf** [comment]
Ok get it.

Btw this particular LIKE support request might be interesting because if I
am not wrong, any character is allowed in the WinGet ID. Hence it cannot be
decided if % or * is wildcard or just an ordinary char 🙂. Some extra
parameter to "enable" wildcard matching with a warning that such issue can
occur could solve this...

But at least for now there seems to be no packages with LIKE special chars in their name...
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/2930419/59d2d87d-0b65-45fa-bde8-2b417f5584f7)


Dne út 20. 2. 2024 20:23 uživatel Romain ***@***.***> napsal:

**Comment by ztrhgf** [comment]
Unfortunately, because I have already one pull request and because of my poor git knowledge I am unable to separate my new commit that supports LIKE from it. I will write the solution here...

My commit that solves this https://github.com/Romanitho/Winget-AutoUpdate/commit/721f22f722515ad20e17deafaf90e1a99800abd6

In Winget-Upgrade.ps1 one line has to be changed from:
```
if (-not ($toSkip -contains $app.Id) -and $($app.Version) -ne "Unknown") {
```

to:
```
if (-not ($toSkip | ? { $app.Id -like $_ }) -and $($app.Version) -ne "Unknown") {
```

And same very similar approach would be for whitelisting IF statement.
